paimon-core/src/main/java/org/apache/paimon/operation/AbstractFileStoreScan.java

```diff
@@ -195,6 +196,70 @@ public FileStoreScan withManifestCacheFilter(ManifestCacheFilter manifestFilter)
 
     @Override
     public Plan plan() {
+        Pair<Long, List<ManifestEntry>> planResult =
+                doPlan(
+                        // how to process entry files
+                        entries ->
+                                entries.parallelStream()
```

Comment:
Can we also put these logical into `doPlan`?
Only `filterByStats` is special, we can just `instanceof ManifestEntry` in `filterByStats`?